data/useThreadById.ts

```diff
@@ -1,10 +1,16 @@
 import useSWR from 'swr';
 import axios from 'axios';
-import { CLUB_SERVER_ROOT } from '../constants';
+import { queryClientForThreadById } from 'helpers/queries';
 
-const getKey = (id: string) => {
-  if (!id) return null;
-  return `${CLUB_SERVER_ROOT}/dagora/thread/${id}?isTestnet=${process.env.NEXT_PUBLIC_IS_TESTNET ? true : false}`;
+const getKey = async (id: string) => {
+  try {
+    if (!id) return null;
+    const parseId = parseInt(id);
+    const result = await queryClientForThreadById(parseId, process.env.NEXT_PUBLIC_IS_TESTNET ? true : false);
+    return result;
+  } catch (err) {
+    return null;
```

Comment:
key should return unique url value , here in this case it should return id only. 
fetcher function will be replaced by queryClientForThreadById, we can now check for testnet inside that function. 

Please complete the learning task for react-swr as well to understand this better.

Reply:
Working on it